### PR TITLE
refactor: User service refactor, remove user mysql client dependency.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
User service refactor, remove user mysql client dependency.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=325037&iterationID=1337&pId=325902&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   User service refactor, remove user mysql client dependency.           |
| 🇨🇳 中文    |   移除 mysql  依赖          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/core/user/user.proto [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+message GetUserRequest {
+    string userID = 1;
+}
+
+message GetUserResponse {
+    User data = 1;
+}
+
+message User {
+    string ID = 1 [json_name = "user_id"];
```
Comment (sfwn) [inline api/proto/core/user/user.proto:38]: not formatted.

File: internal/apps/dop/component-protocol/components/issue-manage/issueTable/render.go [hunk 2]
```diff
@@ -214,7 +214,7 @@ type AssigneeOperationData struct {
 type ComponentAction struct {
 	labels  []apistructs.ProjectLabel
 	isGuest bool
-	userMap map[string]apistructs.UserInfo
+	userMap map[string]string
```
Comment (sfwn) [inline internal/apps/dop/component-protocol/components/issue-manage/issueTable/render.go:217]: any corresponding pb struct here

File: api/proto/core/user/user.proto [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	string name = 2 [json_name = "username"];
+	string nick = 3 [json_name = "nickname"];
+	string avatarURL = 4 [json_name = "avatar_url"];
+    string phone = 5 [json_name = "phone_number"];
+	string email = 6;
+	string state = 7;
+}
+
+message FindUsersRequest {
+    repeated string ids = 1;
```
Comment (sfwn) [inline api/proto/core/user/user.proto:48]: use IDs and set json_name tag.

File: internal/core/user/common/interface.go [hunk 4]
```diff
@@ -14,11 +14,10 @@
 
 package common
 
-import "github.com/erda-project/erda/apistructs"
+import (
+	"github.com/erda-project/erda-proto-go/core/user/pb"
+)
 
 type Interface interface {
```
Comment (sfwn) [inline internal/core/user/common/interface.go:21]: I found you used `pb.UserServiceServer` directly in all related places, not `common.Interface`. So maybe we can delete this unused Interface.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/core/user/user.proto (hunk 1 above)
Comment (shuofan) [reply api/proto/core/user/user.proto:38]: done

File: api/proto/core/user/user.proto (hunk 3 above)
Comment (shuofan) [reply api/proto/core/user/user.proto:48]: done

File: internal/core/user/common/interface.go (hunk 4 above)
Comment (shuofan) [reply internal/core/user/common/interface.go:21]: It is used as a wrapper for initial.
 in internal/core/user/provider.go

File: internal/apps/dop/component-protocol/components/issue-manage/issueTable/render.go (hunk 2 above)
Comment (shuofan) [reply internal/apps/dop/component-protocol/components/issue-manage/issueTable/render.go:217]: refactor as only user name is needed.

**Review by sfwn — APPROVED** [review]

**Pull request merged**